src/postgres_mcp/server.py

```diff
@@ -14,238 +12,242 @@
+    #    return await func(*args, **kwargs)
+    # except Exception as e:
+    #     print(f"Error in {func.__name__}: {e}", file=sys.stderr)
+    #     raise
+
+    return wrapper
+
+
+@mcp.resource(
+    name="list_resources",
+    uri="postgres://resources",
+    description="List available resources, such as database tables and extensions",
+)
+@wrap_with_try_and_reraise
+async def list_resources() -> list[types.Resource]:
+    """List available database tables as resources."""
+    sql_driver = get_safe_sql_driver()
+    rows = sql_driver.execute_query(
+        "SELECT table_name FROM information_schema.tables WHERE table_schema = 'public'"
+    )
+    tables = [row.cells["table_name"] for row in rows] if rows else []
+
+    base_url = "postgres://"
+
+    resources = [
+        types.Resource(
+            uri=AnyUrl(f"{base_url}/{table}/schema"),
+            name=f'"{table}" database schema',
+            description=f"Schema for table {table}",
+            mimeType="application/json",
+        )
+        for table in tables
     ]
 
+    extensions_uri = AnyUrl(f"{base_url}/extensions")
+    resources.append(
+        types.Resource(
+            uri=extensions_uri,
+            name="Installed PostgreSQL Extensions",
+            description="List of installed PostgreSQL extensions in the current database.",
+            mimeType="application/json",
+        )
+    )
+
+    return resources
+
+
+@mcp.resource(
+    name="list_extensions",
+    uri="postgres://extensions",
+    description="List available and installed extensions",
+)
+@wrap_with_try_and_reraise
+async def extensions_resource() -> str:
+    """Get information about installed PostgreSQL extensions."""
+    sql_driver = get_safe_sql_driver()
+    rows = sql_driver.execute_query(
+        """
+        SELECT
+            pae.name AS extname,
+            CASE WHEN pe.extversion IS NOT NULL THEN true ELSE false END AS installed,
+            pe.extversion AS installed_version,
+            pae.default_version,
+            pae.comment
+        FROM
+            pg_available_extensions pae
+        LEFT JOIN
+            pg_extension pe
+            ON pae.name = pe.extname
+        ORDER BY
+            pae.name;
+        """
+    )
+    extensions = [row.cells for row in rows] if rows else []
+    return str(extensions)
+
+
+@mcp.resource(
+    name="list_table_columns",
+    uri="postgres://{table_name}/schema",
+    description="Show columns for the table",
+)
+@wrap_with_try_and_reraise
+async def table_schema_resource(table_name: str) -> str:
+    """Get schema information for a specific table."""
+    sql_driver = get_safe_sql_driver()
+    rows = SafeSqlDriver.execute_param_query(
+        sql_driver,
+        """
+        SELECT column_name, data_type
+        FROM information_schema.columns
+        WHERE table_name = {}
+        """,
+        [table_name],
+    )
+    columns = [row.cells for row in rows] if rows else []
+    return str(columns)
+
+
+@mcp.tool()
```

Comment:
The description is missing from this `tool` and all other tools marked with `mcp.tool`

Reply:
Added.